File: DI String Match/DI String Match.java
class Solution {
    public int[] diStringMatch(String S) {
        int[] res=new int[S.length()+1];
        for(int i=0,l=0,h=S.length();i<=S.length();++i)
            res[i]=i==S.length()||S.charAt(i)=='I'?l++:h--;
        return res;
    }
}

File: Solutions/DI String Match/DI String Match.cpp
class Solution {
public:
    vector<int> diStringMatch(string S) {
        vector<int> res;
        for(int i=0,l=0,h=S.size();i<=S.size();++i)
            res.push_back(i==S.size()||S[i]=='I'?l++:h--);
        return res;
    }
};